src/lib/components/downloadButton/index.tsx

```diff
@@ -14,20 +13,22 @@ export interface DownloadButtonProps {
 const InitialButton = ({ onDownload }: { onDownload: () => void }) => (
   <Button
     className={`w100 ${styles.button}`}
-    buttonTitle="Download"
-    leftIcon={{ src: downloadIcon, alt: 'download arrow icon' }}
+    leftIcon={<DownloadIcon />}
```

Comment:
I'm still wondering if having specific `leftIcon` and `rightIcon` props are necessary?

If we are allowing `children` wouldn't it be enough to pass the icons there?

E.g.
```tsx
 <Button
    className={`w100 ${styles.button}`}
    onClick={onDownload}
    data-cy="download-documents-button"
  >
  <DownloadIcon className="mr8"/> Download
</Button>
```
Or something similar?

It makes sense for input components, since adding stuff inside the input is very annoying, but with buttons there is not much value being added. Also moving towards this direction makes the component less flexible.

What do you think?

Reply:
For me it's about having a consistent design for buttons without having to worry about it when implementing buttons.

Two examples I can think of are: 
How can you make sure that Icons always have the same size and spacing between icons and text?
Or if we have a `8` spacing and we want to switch to 16 in the future, we need to go through every component usage to make it have a standard logic, which IMO, should be contained within the button.

Also `children` is still accepting `ReactNode` and still allows you to add anything you want in there. WDYT?

Reply:
That's a good point 🤔 
I can think of a scenario when we need an icon with a spacing other than `8`, but I guess that shouldn't happen that often, and in those cases we can fall back to the children.

I'm going to resolve this conversation to keep the PR clean and tidy.
Thanks for your insights!